filecoin-proofs/src/param.rs

```diff
@@ -75,18 +81,42 @@ pub fn load_parameter_map() -> Result<ParameterMap> {
     }
 }
 
-pub fn save_parameter_map(map: ParameterMap) -> Result<()> {
-    let file = File::create(PARAMETER_JSON_PATH)?;
+pub fn save_parameter_map(map: &ParameterMap, path: &PathBuf) -> Result<()> {
+    let file = File::create(path)?;
     let writer = BufWriter::new(file);
     serde_json::to_writer_pretty(writer, &map)?;
 
     Ok(())
 }
 
-pub fn publish_parameter_file(parameter: String) -> Result<String> {
+pub fn parameter_file_path(parameter: String) -> Result<PathBuf> {
     let mut path = parameter_cache_dir();
     path.push(parameter);
 
+    if path.exists() {
+        Ok(path)
+    } else {
+        Err(err_msg(ERROR_PARAMETER_FILE))
+    }
+}
+
+pub fn get_parameter_digest(parameter: String) -> Result<String> {
+    let path = parameter_file_path(parameter)?;
+    let mut file = File::open(path)?;
+    let mut buffer = Vec::new();
+    let _ = file.read_to_end(&mut buffer);
```

Comment:
this reads the whole file into memory, which is not needed, you should be able to do sth like this

```rust
let mut file = File::open(parameter)?;
let mut hasher = Blake2b::new();
std::io::copy(&mut file, &mut hasher)?;
let hash = hasher.result();
```